src/components/cl/basic/cl_basic_lib.c

```diff
@@ -28,43 +28,64 @@ UCC_CLASS_CLEANUP_FUNC(ucc_cl_basic_lib_t)
 
 UCC_CLASS_DEFINE(ucc_cl_basic_lib_t, ucc_cl_lib_t);
 
-ucc_status_t ucc_cl_basic_get_lib_attr(const ucc_base_lib_t *lib,
-                                       ucc_base_attr_t *base_attr)
+static inline void check_tl_lib_attr(const ucc_base_lib_t *lib,
+                                             ucc_tl_iface_t *tl_iface,
+                                             ucc_cl_lib_attr_t *attr)
 {
-    ucc_cl_lib_attr_t     *attr   = ucc_derived_of(base_attr, ucc_cl_lib_attr_t);
-    ucc_cl_lib_t          *cl_lib = ucc_derived_of(lib, ucc_cl_lib_t);
-    ucc_status_t           status;
-    ucc_tl_lib_attr_t      ucp_tl_attr, nccl_tl_attr;
-    ucc_component_iface_t *ucp_iface, *nccl_iface;
-    ucc_tl_iface_t        *tl_ucp_iface, *tl_nccl_iface;
-    attr->tls = &cl_lib->tls;
-    ucp_iface = ucc_get_component(&ucc_global_config.tl_framework, "ucp");
-    if (!ucp_iface) {
-    	cl_error(lib, "failed to get UCP component");
-    	return UCC_ERR_NO_RESOURCE;
-    }
-    tl_ucp_iface = ucc_derived_of(ucp_iface, ucc_tl_iface_t);
-    memset(&ucp_tl_attr, 0, sizeof(ucc_tl_lib_attr_t));
-    status = tl_ucp_iface->lib.get_attr(NULL, &ucp_tl_attr.super);
+    ucc_tl_lib_attr_t tl_attr;
+    ucc_status_t      status;
+
+    memset(&tl_attr, 0, sizeof(tl_attr));
+    status = tl_iface->lib.get_attr(NULL, &tl_attr.super);
     if (UCC_OK != status) {
-        cl_error(lib, "failed to query cl lib attributes");
-        return status;
+        cl_warn(lib, "failed to query tl %s lib attributes",
```

Comment:
why not return error?

Reply:
right, no reason - reverted it back. At first i thought that maybe we can still proceed, but now i agree. If a component failed to report its caps we can't use it anyways. So, reporting error.